ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java

```diff
@@ -429,6 +435,59 @@ private boolean canAuthenticate(ConnectionStringBuilder engineCsb) {
         return true;
     }
 
+    private void assertUrlCompare(String connectionDataSource, String clusterUrl, boolean autoCorrectEndpoint, boolean isQueued) {
+        if (!autoCorrectEndpoint || clusterUrl.contains(INGEST_PREFIX) || isReservedHostname(clusterUrl)) {
+            String host = clusterUrl.replaceFirst("https" + PROTOCOL_SUFFIX, "");
+            if (InetAddressUtils.isIPv6Address(host)) {
+                String compareString = clusterUrl.replaceFirst(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + '[') + ']';
+                assertEquals(compareString.toLowerCase(), connectionDataSource);
+            } else {
+                assertEquals(clusterUrl, connectionDataSource);
+            }
+        } else {
+            String compareString = isQueued ? clusterUrl.replaceFirst(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + INGEST_PREFIX) : clusterUrl;
+            assertEquals(compareString, connectionDataSource);
+        }
+    }
+
+    private static Stream<Arguments> provideStringsForAutoCorrectEndpointTests() {
+        return Stream.of(
```

Comment:
Positive examples? Edge cases?

Testing against System.getenv("ENGINE_CONNECTION_STRING") is not recommended, have just tests that pass

Reply:
Not sure what you're referring to - these are edge cases which we discussed here on the PR
furthermore - all of these tests pass, so again not quite sure what you are referring to

Reply:
I meant tests that pass the check, and have their url be autocorrected.

Reply:
so isn't System.getenv("ENGINE_CONNECTION_STRING") just that? or would you prefer it being a hardcoded URL instead?

Reply:
There are many cases that do have a replacement but close to cases to aren't (for example - an address with a lot of numbers that isn't an ip address). 
You only test ONCE, on an env variable provided by the user, which could be anything, even one of the ones that is exempted.

Test against known urls, so you can know what the result should be.

Write at least 10 cases that do the substitution, with something different for each of them.